### DISCUSSION TITLE
Npm doesn’t use my custom private registry

### DISCUSSION THREAD

**Discussion by azro352** [renovatebot/renovate#24951] [Request Help] [answered]
Self hosted renovate on self hosted gitlab, docker 37.2.0-full

I'Ve defined my private registry , the same I have defined the ones for docker and python

```js
module.exports ={
  packagesRules : [
     {matchDataSources :["npm"], registryUrls : ["https://myregistry"]}
]
}
```

In the logs 

- It well associate the registry to npm packages and even find the latest version available in the registry 
- the command `npm install --package-lock-only --no-audit --ignore-scripts thepackage` fails stating it failed to reach npmjs.org but it shouldn't try it 

What do I miss ?

**Comment by rarkins** [comment]
Try adding the `npmrc` setting too

**Reply by azro352** [reply]
With value `registry=https://...` or just the url ?

**Reply by rarkins** [reply]
Yes, it must be a valid `.npmrc` string, like this: `"npmrc": "registry=https://myregistry"`

**Reply by azro352** [reply]
And if I want to put multi line content ? Like add a `strict-ssl=false`

**Reply by rarkins** [reply, marked as answer]
Use `\n` to separate
